integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/projection/SearchProjectionIT.java

```diff
@@ -122,20 +122,23 @@ public void references() {
 				scope.projection().entityReference().toProjection();
 		SearchProjection<DocumentReference> objectProjection =
 				scope.projection().entity().toProjection();
+		SearchProjection<Object> idProjection =
+				scope.projection().id().toProjection();
 
 		query = scope.query()
 				.select(
 						documentReferenceProjection,
 						entityReferenceProjection,
-						objectProjection
+						objectProjection,
+						idProjection
 				)
 				.where( f -> f.matchAll() )
 				.toQuery();
 		assertThatQuery( query ).hasListHitsAnyOrder( b -> {
-			b.list( document1Reference, document1Reference, document1Reference );
-			b.list( document2Reference, document2Reference, document2Reference );
-			b.list( document3Reference, document3Reference, document3Reference );
-			b.list( emptyReference, emptyReference, emptyReference );
+			b.list( document1Reference, document1Reference, document1Reference, DOCUMENT_1 );
+			b.list( document2Reference, document2Reference, document2Reference, DOCUMENT_2 );
+			b.list( document3Reference, document3Reference, document3Reference, DOCUMENT_3 );
+			b.list( emptyReference, emptyReference, emptyReference, EMPTY );
```

Comment:
I think this feature it warrants additional testing, especially if you follow my recommendation of implementing it similarly to the ID predicate.

I think we'd need tests inspired from these, at least:

* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionSingleValuedBaseIT#simple`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionSingleValuedBaseIT#noClass`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionSingleValuedBaseIT#duplicated`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionTypeCheckingAndConversionIT#validSuperClass`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionTypeCheckingAndConversionIT#invalidProjectionType_projectionConverterEnabled`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionTypeCheckingAndConversionIT#multiIndex_withCompatibleIndex_projectionConverterEnabled`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionTypeCheckingAndConversionIT#multiIndex_withIncompatibleIndex_projectionConverterEnabled`
* `org.hibernate.search.integrationtest.backend.tck.search.projection.FieldSearchProjectionTypeIndependentIT#nullClass` (this one doesn't need its own test class in the case of the ID projection, obviously)

I'd suggest you keep the organization with one `*IT` class per concern, since it's done that way in many other tests.